### PR TITLE
Improved testsuites hierarchy

### PR DESCRIPTION
**Pull request by jonbell-lot23**
**Summary**

Last week I pushed [this PR](https://github.com/replayio/devtools/pull/9188) to improve visual hierarchy, but it failed in some situations. That led to [this bug](https://linear.app/replay/issue/FE-1480/test-suites-font-size-is-off) and Brian reverted [here](https://github.com/replayio/devtools/commit/67109276d27b998d81d5854bf46e350021bd2035).

This commit addresses the issue more carefully, and I've tested it on these links (all localhost):

[Link one](http://localhost:8080/recording/cypresse2eadding-spects--2510b505-2ae9-4009-baea-4a866cd2a69b?point=7463926734776456180313999007547641&time=3897&focusRegion=eyJiZWdpbiI6eyJwb2ludCI6Ijc0NjM5MjY3MzQ3NzY0NTYxODAzMTM5OTkwMDc1NDc2NDEiLCJ0aW1lIjozODk3fSwiZW5kIjp7InBvaW50IjoiMTM2Mjk3NzkyNTUxMzQ2MTk2MTMyNTkxMjY0MzU0ODAwOTgiLCJ0aW1lIjo2MDMyfX0%253D)

[Link two](http://localhost:8080/recording/cypresse2eediting-spects--246f386c-f890-42fe-aa08-ac1fd756704c?point=28557632726786868294881658399621120&time=12169)

[Link three](http://localhost:8080/recording/cypresstestsuibankaccountsspects--f7ef4ea0-5f50-474a-aa70-f89cabe5a0da?point=14603334920664384764506418295342008&time=9415&focusRegion=eyJiZWdpbiI6eyJwb2ludCI6Ijc0NjM5MjY3MzkxMzkyNzI1NjI3NTY5NTI5MjMzMDgyOTYiLCJ0aW1lIjo0ODQzfSwiZW5kIjp7InBvaW50IjoiMjMwNDA4MTczMTg0NTk3NzEzMzc0OTE5MDg2MDQ3MjYxODEiLCJ0aW1lIjoxMjUwNn19)

[Link four](http://localhost:8080/recording/cypressintegrationallsortingspecjs--229a01dd-c01f-45cc-9ad7-2c08ce6a12a0?point=2271629876170247722668778961502515&time=1267)

[Link five](http://localhost:8080/recording/cypressintegrationallfiltersspects--cefb8a85-197f-49e0-bea4-3669d383b2df?point=2596148429826941608430510891073820&time=1400)

Old:
image

New:
image

### CODE REVIEW

File: src/ui/components/TestSuite/views/TestMetadata/Panel.module.css [hunk 1]
```diff
@@ -18,7 +18,6 @@
   flex: 1 1 auto;
   font-size: var(--font-size-large);
   color: var(--body-color);
-  overflow: hidden;
```
Comment (bvaughn) [inline src/ui/components/TestSuite/views/TestMetadata/Panel.module.css:21]: What motivated this change? It will cause a regression so I think it should be reverted.

### With overflow
Screen Shot 2023-05-15 at 9 23 41 AM

### Without overflow
Screen Shot 2023-05-15 at 9 23 32 AM

File: src/ui/components/TestSuite/views/TestMetadata/TestItemTree.tsx [hunk 2]
```diff
@@ -57,28 +57,29 @@ function TestTree({
...
 }: {
-  isNested?: boolean;
+  isTopScope?: boolean;
   scope?: string | null;
   selectTestItem: (testItem: ProcessedTestItem) => void;
   testTree: TestTree;
 }) {
-  const parentScope = scope;
+  const className = scope ? styles.TreeNested : styles.Tree;
+  const scopeClassName = isTopScope ? styles.TopScope : styles.Scope;
```
Comment (bvaughn) [inline src/ui/components/TestSuite/views/TestMetadata/TestItemTree.tsx:75]: What's the purpose of this change?

There is no `TopScope` class in the CSS module for this component, so this looks like a mistake.

**Review by bvaughn — CHANGES_REQUESTED** [review]

File: src/ui/components/TestSuite/views/TestMetadata/Panel.module.css (hunk 1 above)
Comment (jonbell-lot23) [reply src/ui/components/TestSuite/views/TestMetadata/Panel.module.css:21]: True. I was trying to address an issue where the descenders (for example, the lower part of the lowercase g) were getting cropped. This fixes that issue, but the side effect you point out is bad.

So instead I'm going to force a line height.

File: src/ui/components/TestSuite/views/TestMetadata/TestItemTree.tsx (hunk 2 above)
Comment (jonbell-lot23) [reply src/ui/components/TestSuite/views/TestMetadata/TestItemTree.tsx:75]: Yup, I was going down this road in order to style the parent differently from nested items, but changed my mind. Then I forgot to remove this code, so this was indeed a mistake. Thanks!

File: src/ui/components/TestSuite/views/TestMetadata/Panel.module.css (hunk 1 above)
Comment (bvaughn) [reply src/ui/components/TestSuite/views/TestMetadata/Panel.module.css]: If that was your concern, rather than forcing a line-height (which can cause other layout issues) – just change this to `overflow-x: hidden` ?

File: src/ui/components/TestSuite/views/TestMetadata/TestItemTree.module.css [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
 .Tree[data-nested] {
-  padding-left: 0.75rem;
 }
 
 .List {
   display: flex;
   flex-direction: column;
   gap: 0.5rem;
-  padding-left: 0.75rem;
+  margin-bottom: 1rem;
```
Comment (bvaughn) [inline src/ui/components/TestSuite/views/TestMetadata/TestItemTree.module.css:12]: This is subjective 😄  but I don't dig this margin change:
Screen Shot 2023-05-16 at 8 59 39 AM

Vertical space between our items isn't consistent, and there's a lot of space now between lists.

**Review by bvaughn — APPROVED** [review]

File: src/ui/components/TestSuite/views/TestMetadata/TestItemTree.module.css (hunk 3 above)
Comment (jonbell-lot23) [reply src/ui/components/TestSuite/views/TestMetadata/TestItemTree.module.css:12]: Having larger spaces between lists is appropriate (in my subjective opinion) because they're the main thing you're scanning for.

It's not exactly the same as line height versus gaps between paragraphs but it's in the same ballpark.

But I take your point and there are consistencies we can probably find. Will merge this and file a separate task for margins.
